scripts/octoapp.sh

```diff
@@ -0,0 +1,385 @@
+    else
+      log_error "Internal error. moonraker_count of '${moonraker_count}' not equal or grater than one!"
+      return 1
+    fi  # (( moonraker_count == 1 ))
+  fi  # (( allowed_octoapp_count > 0 ))
+
+  # Special case for one moonraker instance with OctoApp already installed.
+  # If the user selects the install option again, they might be trying to recover the install
+  # or complete a printer link they didn't finish in the past.
+  # So in this case, we will allow them to run the install script again, since it's safe to run
+  # if the service is already installed, it will repair any missing issues.
+  if (( allowed_octoapp_count == 0 && moonraker_count == 1 )); then
+    local yn
+    while true; do
+      echo "${yellow}OctoApp is already installed.${white}"
+      echo "It is safe to run the install again to repair any issues or if the printer isn't linked, run the printer linking logic again."
+      echo ""
+      local question="Do you want to run the OctoApp recovery or linking logic again?"
+      read -p "${cyan}###### ${question} (Y/n):${white} " yn
+      case "${yn}" in
+        Y|y|Yes|yes|"")
+          select_msg "Yes"
+          break;;
+        N|n|No|no)
+          select_msg "No"
+          abort_msg "Exiting OctoApp setup ...\n"
+          return;;
+        *)
+          error_msg "Invalid Input!";;
+      esac
+    done
+    # The user responded yes, allow the install to run again.
+    allowed_octoapp_count=1
+  fi
+
+  # If there's something to install, do it!
+  if (( allowed_octoapp_count > 0 )); then
+
+    (( new_octoapp_count > 1 )) && status_msg "Installing ${new_octoapp_count} OctoApp instances ..."
+    (( new_octoapp_count == 1 )) && status_msg "Installing OctoApp ..."
+
+    # Ensure the basic system dependencies are installed.
+    local dep=(git dfu-util virtualenv python3 python3-pip python3-venv)
+    dependency_check "${dep[@]}"
+
+    # Close the repo
+    clone_octoapp "${OCTOAPP_REPO}"
+
+    # Call install with the correct args.
+    local instance_cfg_dirs
+    read -r -a instance_cfg_dirs <<< "$(get_instance_folder_path "config")"
+    echo instance_cfg_dirs[0]
+
+    if (( moonraker_count == 1 )); then
+      "${OCTOAPP_DIR}/install.sh" "${instance_cfg_dirs[0]}/moonraker.conf"
+    elif (( moonraker_count > 1 )); then
+      local j=${existing_octoapp_count}
+
+      for (( i=1; i <= new_octoapp_count; i++ )); do
+        "${OCTOAPP_DIR}/install.sh" "${instance_cfg_dirs[${j}]}/moonraker.conf"
+        j=$(( j + 1 ))
+      done && unset j
+    fi # (( moonraker_count == 1 ))
+  fi  # (( allowed_octoapp_count > 0 ))
+}
+
+function clone_octoapp() {
+  local repo=${1}
+
+  status_msg "Cloning OctoApp..."
+  ### force remove existing repos
+  [[ -d "${OCTOAPP_DIR}" ]] && rm -rf "${OCTOAPP_DIR}"
+
+  cd "${HOME}" || exit 1
+  if ! git clone "${repo}" "${OCTOAPP_DIR}"; then
+    print_error "Cloning OctoApp from\n ${repo}\n failed!"
+    exit 1
+  fi
+}
```

Comment:
This function is declared twice in this file. This is coming form the OctoEverywhere script, where this function is also declared twice. We can simply remove that one here.

Reply:
Removed. I ping Quinn about the duplicate function in OE, but as both are identical there is not much harm done